### PR TITLE
Feat/smart contracts update

### LINKED ISSUE #483: Smart contracts update
Update smart contract addresses and ABIs for the new bridge

- [ ] Add the new ABI as a new abi in the smart-contracts repo
- [ ] Create a new class for the smart contracts (we can't edit existing as there will be a period of time where we will need to support both). Base this on the existing class.
- [ ] Add new withdrawals hook to support additional parameters now needed
- [ ] Add logic to use old or new bridge based on an environment variable

[ref](https://vegaprotocol.slack.com/archives/C026LG6B83E/p1653557426863809)

### PR DESCRIPTION
**Pull request by dexturr**
# Related issues 🔗

Closes #483 

# Description ℹ️

Adds the ability to use the new or old bridge contracts. The old one will need ot be used on the token site for mainnet until after mainnet has the new smart contracts enabled.

### CODE REVIEW

File: apps/token/src/routes/withdraw/index.tsx [hunk 1]
```diff
@@ -149,7 +150,11 @@ export const WithdrawContainer = ({ currVegaKey }: WithdrawContainerProps) => {
           </Callout>
         </div>
       )}
-      <WithdrawManager assets={data.assets || []} accounts={accounts} />
+      <WithdrawManager
+        assets={data.assets || []}
+        accounts={accounts}
+        useNewContract={Flags.USE_NEW_CONTRACTS}
```
Comment (mattrussell36) [inline apps/token/src/routes/withdraw/index.tsx:156]: chore (blocking): Can we rename this one too `isNewContract`

**Review by mattrussell36 — CHANGES_REQUESTED** [review]
Couple things to change here. Looks good, though I don't much like how we use collateral bridge and collateral bridge new. I think I would prefer to rename the current one to collateral-bridge-deprecated. Also I wonder if we can look into Jeremy's suggestion on slack

File: libs/smart-contracts/src/contracts/collateral-bridge.ts [hunk 2]
```diff
@@ -4,6 +4,7 @@ import abi from '../abis/erc20_bridge_abi.json';
 
 export class CollateralBridge {
   public contract: ethers.Contract;
+  public isNewContract = false;
 
   constructor(
```
[the remark this answers is not in the document]
Comment (dexturr) [reply libs/smart-contracts/src/contracts/collateral-bridge.ts:9]: I think I'd prefer to add a deprecated decorator, which we can use everywhere. Will do this 👍

**Review by mattrussell36 — APPROVED** [review]

File: libs/withdraws/src/lib/use-withdraw.spec.tsx [hunk 3]
```diff
@@ -82,7 +82,7 @@ beforeEach(() => {
   };
   mockERC20Approval = {
     request: {
-      query: ERC20_APPROVAL_QUERY,
+      query: ERC20_APPROVAL_QUERY_NEW,
```
Comment (sam-keen) [inline libs/withdraws/src/lib/use-withdraw.spec.tsx:85]: Chore (non-blocking): ideally the tests should cover both contracts

**Review by sam-keen — APPROVED** [review]

**Review by MadalinaRaicu — APPROVED** [review]

**Pull request merged**